front/src/pages/AddItem/AddItem.tsx

```diff
@@ -8,24 +8,32 @@ import {
 import { importanceValues } from '../../constants';
 import Loader from '../../components/Loader';
 
+const initialValues = {
+  todoTitle: '',
+  todoImportance: Object.keys(importanceValues)[0],
+}
+
 const AddItem = () => {
+  const [todoTitle, setTodoTitle] = useState(initialValues.todoTitle);
+  const [todoImportance, setTodoImportance] = useState(initialValues.todoImportance);
   const {
     mutateAsync: addTodo,
     isLoading
   } = useAddTodo();
-  const addFormRef = useRef<HTMLFormElement>(null);
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => setTodoTitle(e.target.value)
+  const handleImportanceChange = (e: React.ChangeEvent<HTMLSelectElement>) => setTodoImportance(e.target.value)
 
   const resetForm = () => {
-    addFormRef?.current?.reset();
+    setTodoTitle(initialValues.todoTitle);
+    setTodoImportance(initialValues.todoImportance);
   }
 
-  const handleFormSubmit = (e: React.SyntheticEvent) => {
+  const handleFormSubmit = (e: React.FormEvent) => {
     e.preventDefault();
-    const target = e.target as HTMLFormElement;
-    const formFields = Object.fromEntries(new FormData(target)) as {[k: string]: string} ;
     const newTodo: TodoToCreate = {
-      title: formFields.todoTitle,
-      importance: formFields.todoImportance as TodoImportance,
+      title: todoTitle,
+      importance: todoImportance as TodoImportance,
     }
     addTodo(newTodo).then(resetForm);
```

Comment:
btw what if `addTodo` fails? we need some error message to be shown

Reply:
@skyboyer 
this logic implemented inside the useAddTodo().
Should I move an error dispatch from useAddTodo() to AddItem component to make it more obvious?

Reply:
no, that's fine